Document: Add SMBIOS additional Information for LoongArch (#34)
loongson/LoongArch-Documentation: docs/LoongArch-Processor-SMBIOS-Spec-EN.adoc

= LoongArch Processor SMBIOS Spec
Loongson Technology Corporation Limited
v1.00
:docinfodir: ../themes
:docinfo: shared
:doctype: book
:toc: left

> This document defines LoongArch processor-specific data block to supplement the upstream definition of SMBIOS structure Type 44 (Processor Additional Information, SMBIOS 3.5.0 or later specification section 7.45).

Reviewer: ```suggestion
This document defines the LoongArch processor-specific data block to supplement the upstream definition of SMBIOS structure Type 44 (Processor Additional Information, section 7.45 in SMBIOS specification 3.5.0 or later).
```

=== Symbols

* *DQWORD* +

> 128-bits (In the SMBIOS specification, WORD is 16-bits, DWORD is 32-bits ,and QWORD is 64-bits).

Reviewer: ```suggestion
128-bits (In the SMBIOS specification, WORD is 16-bits, DWORD is 32-bits, and QWORD is 64-bits).
```

=== Vendor Name register and CPU Name register

> The LoongArch CPU designed by Loongson has two registers representing Machine Vendor Name and CPU Name, both of which are ASCII string values in DQWORD-formatted, respectively located at offsets 0x10 and 0x20 of the IOCSR space.

Reviewer: ```suggestion
The LoongArch CPUs designed by Loongson have two registers representing Machine Vendor Name and CPU Name, both of which are DQWORD-format NUL-terminated ASCII string values, located at offsets 0x10 and 0x20 of the IOCSR space respectively.
```

Author: 这里是第三人称吧，用have不合适吧

Reviewer: 这里要表示的是适用 **所有** 龙芯公司设计的 LoongArch CPU，因此要用复数符合表达习惯

=== LoongArch Processor-specific Block Structure
[%header,cols="^1,^1,^1,^1,3"]
|===
|Offset
|Name
|Length
|Value
^|Description

|00h
|Block Length
|BYTE
|Varies (N)
|Length of Processor-specific Data

|01h
|Processor Type
|BYTE
|Varies
|The processor architecture delineated by this Processor-specific Block (See SMBIOS Table 131). 0x9 means LoongArch32 and 0x10 means LoongArch64.

|02h
|Processor-Specific Data
|N BYTEs
|Varies
|Processor-specific data.

|03h
|Reserved
|BYTE
|Varies
|Reserved.

|04h
|Machine Vendor ID
|DQWORD
|Varies
|The manufacturer vendor ID of the processor. +

> It is equal to the value at the offset 0x10 of the IOCSR space on a Loongson CPU.

Reviewer: 强调下只是语义等价，这样可以给非龙芯公司的方案充足的自由度，也不至于影响互操作性：

```suggestion
It is semantically equivalent to the value at the offset 0x10 of the IOCSR space on a Loongson CPU.
```

> It is semantically qeuivalent to the value at the offset 0x10 of the IOCSR space on a Loongson CPU.

Reviewer: ```suggestion
It is semantically equivalent to the value at the offset 0x10 of the IOCSR space on a Loongson CPU.
```

|14h
|CPU ID
|DQWORD
|Varies
|The CPU ID used for this LoongArch processor manufacturer to mark different CPU types or CPU instances. +

> It is equal to the value at the offset 0x20 of the IOCSR space on a Loongson CPU.

Reviewer: 同上

|24h
|ISA extensions support
|DWORD
|Bit-field

> |The bit field [3:0] indicate support for the existing LoongArch standard ISA extensions. It is modeled on the LoongArch EUEN register (CSR 0x2), and meanings of each bits is the same as defined for the EUEN register. +

Reviewer: `model after` 是固定搭配，另外还有其他的单复数问题

```suggestion
|The bit field [3:0] indicates support for the existing LoongArch standard ISA extensions. It is modeled after the LoongArch EUEN register (CSR 0x2), and meaning of each bit is the same as defined for the EUEN register. +
```

> Any bit of this field is set, indicating that the corresponding ISA extension is supported on this system.

Reviewer: 这个句子翻译过来是“这个位域的任何位都（已经）置位了”，意思不对了。需要改一下。

```suggestion
A set bit means its corresponding ISA extension is supported on this system.
```
